notebooks/11-xarray-intro.md

```diff
@@ -1,185 +1,68 @@
-
-with rasterio.open(data_file) as src:
-    herstappe_array = src.read()
-    herstappe_meta = src.meta
-```
-
-```{code-cell} ipython3
-herstappe_array
-```
-
-As we learnt in the previous lesson, Rasterio returns a Numpy `ndarray`:
-
-```{code-cell} ipython3
-type(herstappe_array)
-```
-
-Numpy supports different `dtype`s (`float`, `int`,...), but all elements of an array do have the same dtype. Note that NumPy auto-detects the data-type from the input.
-
-```{code-cell} ipython3
-herstappe_array.dtype
-```
-
-The data type of this specific array `herstappe_array` is float32. More information on the data types Numpy supports is available in the [documentation](https://numpy.org/devdocs/user/basics.types.html#array-types-and-conversions-between-types).
-
-+++
-
-Converting to another data type is supported by `astype` method. When floats are preferred during calculation:
-
-```{code-cell} ipython3
-herstappe_array.astype(int)
-```
-
-Just as any other object in Python, the `ndarray` has a number of attributes. We already checkes the `dtype` attribute. The `shape` and `ndim` of the array are other relevant attribute:
-
-```{code-cell} ipython3
-herstappe_array.shape, herstappe_array.ndim
-```
-
-Hence, we have three bands with dimensions (227, 447) and data type `float32`. Compare this to the metadata stored in the geotiff file:
-
-```{code-cell} ipython3
-# !gdalinfo ./data/herstappe/raster/2020-09-17_Sentinel_2_L1C_True_color.tiff
+ax = herstappe.plot.imshow(figsize=(12, 5))
+ax.axes.set_aspect('equal')
 ```
 
-The metadata on the dimensions and the datatype correspond, but the spatial information is lost when we only store the Numpy array. While if we use `xarray.DataArray` to wrap this NumPy array, we can keep the data and the spatial context together.
-
-+++
-
-<div class="alert alert-info" style="font-size:120%">
-
-**REMEMBER**: <br>
-
-The [`numpy` package](https://numpy.org/) is the backbone of the scientific Python ecosystem. The `ndarray` provides an efficient data type to store and manipulate raster data, but it does NOT contain any spatial information.
-
-The [11-numpy.ipynb](11-numpy.ipynb) notebook provides more content on using pure numpy with rasterio and how to keep track of the spatial extent (eg when plotting), but for this notebooks, we will use xarray to provide context to our numpy array dimensions.
-
-</div>
-
-+++
-
-Let's take a look at our xarray.DataArray again:
+We can further make the dimension information more useful by renaming the labels of the 'band' dimension:
 
 ```{code-cell} ipython3
+herstappe = herstappe.assign_coords(band=("band", ["red", "green", "blue"]))
 herstappe
 ```
 
-We can further make the dimension information more useful by renaming the labels of the "band" dimension:
+Or just rename the coordinate names itself:
 
 ```{code-cell} ipython3
-herstappe = herstappe.assign_coords(band=("band", ["red", "green", "blue"]))
-herstappe
+herstappe.rename({"x": "longitude", "y": "latitude"})
```

Comment:
It's a nice example to rename this, but the values are not longitude/latitude (since that would mean geographical coordinates, and it are projected coordinates), so the new name is actually incorrect ..

Reply:
Good point! I'll make it easting/northing.